theano/tensor/nnet/nnet.py

```diff
@@ -302,39 +301,28 @@ class SoftmaxGrad(gof.Op):
     def make_node(self, dy, sm):
         dy = tensor.as_tensor_variable(dy)
         sm = tensor.as_tensor_variable(sm)
-        if dy.type.ndim not in (1, 2) \
-                or dy.type.dtype not in tensor.float_dtypes:
-            raise ValueError('dy must be 1-d or 2-d tensor of floats. Got ',
-                             dy.type)
-        if dy.ndim == 1:
-            dy = tensor.shape_padleft(dy, n_ones=1)
-        if sm.ndim == 1:
-            sm = tensor.shape_padleft(sm, n_ones=1)
         return Apply(self, [dy, sm], [sm.type()])
 
     def perform(self, node, input_storage, output_storage):
         dy, sm = input_storage
+        if (dy.shape != sm.shape):
+            raise ValueError('dy and the softmax output should have the same shape.')
```

Comment:
In the old version of the code (The current one of Theano), they check if dy and sm has the same shape in the C code. So, to be consistent with the C code, I though it will be needed to do the same things in the python code.